main.go

```diff
@@ -149,22 +149,32 @@ func handleEvent(event vtypes.BaseEvent, m *event.Manager) (string, string, erro
 
 	topic := convertToTopic(eventType)
 
-	message, _ := json.Marshal(OutboundEvent{
+	message, err := json.Marshal(OutboundEvent{
 		Topic:       topic,
 		Category:    category,
-		CreatedTime: gotEvent.CreatedTime,
 		UserName:    gotEvent.UserName,
+		CreatedTime: gotEvent.CreatedTime,
+		MoRef:       ref,
 	})
+	if err != nil {
+		return "", "", errors.Wrap(err, "error marshaling outboundevent")
+	}
+
+	log.Println(string(message))
 
 	return topic, string(message), nil
 }
 
+// OutboundEvent is the JSON object sent to subscribed functions.
+// If the ManagedObjectReference for an event cannot be retrieved, MoRef will be nil and thus not marshaled into the JSON OutboundEvent.
+// It's the receivers responsibility to check whether MoRef contains valid data (or nil/null, depending on the JSON unmarshaling logic used).
 type OutboundEvent struct {
 	Topic    string `json:"topic,omitempty"`
 	Category string `json:"category,omitempty"`
 
-	UserName    string    `json:"userName,omitempty"`
-	CreatedTime time.Time `json:"createdTime,omitempty"`
+	UserName    string                         `json:"userName,omitempty"`
+	CreatedTime time.Time                      `json:"createdTime,omitempty"`
+	MoRef       *vtypes.ManagedObjectReference `json:"moref,omitempty"`
```

Comment:
Should we go for `ManagedObjectReference`?

Reply:
Reason I choose a pointer is that if *ManagedObjectReference is nil, it won't be marshaled. If it's a struct, it will be marshaled, bypassing "omitempty" (https://play.golang.org/p/z6cwin69PDC). Thus, on the receiver side it's easier to unmarshal and check whether the field exists/is nil (depending on the language and JSON implementation) then dealing with `types.ManagedObjectReference{}` being empty.

Reply:
Sorry I didn't explain that well. Should we rename `MoRef` to the longer variant?

Reply:
Will change to:

```go
// OutboundEvent is the JSON object sent to subscribed functions
// If the ManagedObjectReference for an event cannot be retrieved, it will be nil and thus not marshaled into the JSON OutboundEvent
// It's the receivers responsibility to check whether managedObjectReference key is present in the JSON message payload
type OutboundEvent struct {
	Topic    string `json:"topic,omitempty"`
	Category string `json:"category,omitempty"`

	UserName               string                         `json:"userName,omitempty"`
	CreatedTime            time.Time                      `json:"createdTime,omitempty"`
	ManagedObjectReference *vtypes.ManagedObjectReference `json:"managedObjectReference,omitempty"`
}
```

Reply:
- [ ] Update Go and Python examples on changed JSON key for MoRef